### DISCUSSION TITLE
How do i remove a the uploaded file list after submission?

### DISCUSSION THREAD

**Discussion by mattjaf** [ant-design/ant-design#40774] [General]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/0kqgzs)

### Steps to reproduce

after submitting the a file to the upload component with the file list, It still shows the files there, how can i remove them?

### What is expected?

that I would beable to progromatically remove them

### What is actually happening?

they still there after a form submission

| Environment | Info |
| --- | --- |
| antd | 4.22.7 |
| React | 17.02 |
| System | windows 11 |
| Browser | chrome |

---
It would be nice to beable to post a screenshot...
do you guys have slack or discord?

**Comment by mattjaf** [comment]
oh theres a fileList prop,

is there a way to add a name to the file, or is that something i gotta put in a modal?

also is there an example with all the props in actions?

**Reply by arshad657** [reply]
Is there any solutions you got @mattjaf  ?

**Reply by jiad-saegus** [reply]
This is how i did it ( @arshad657  ):

states and helper functions:
```JSX
const [fileList, setFileList] = useState<UploadFile[]>([]) // this will hold the files

const onChange: UploadProps["onChange"] = ({ fileList: newFileList }) => {
  setFileList(newFileList);
};

const clearUpload = () => {
  setFileList([]);
};

const onFinish = async (values: object) => {
  // your submit code
  clearUpload();
}
```


component:
```JSX
<Upload
  beforeUpload={() => false}
  ref={uploadRef}
  fileList={fileList}
  onChange={onChange}
>
  <Button icon={<BarsOutlined />}>Click to Upload</Button>
</Upload>
```
